app/tests/test_amity.py

```diff
@@ -0,0 +1,150 @@
+
+    def test_adding_existing_person(self):
+        self.amity.all_people.append("Paul Upendo")
+        self.assertTrue(self.amity.add_person("Paul", "Upendo", "FELLOW", "N")
+                        == "Ooops! Paul Upendo already exists in the system.")
+
+    def test_add_person_to_system(self):
+        self.assertEqual(self.amity.add_person("John", "Waria", "FELLOW", "Y"),
+                         'Person has been successfully added')
+
+    def test_fellow_data_persisted(self):
+        self.amity.add_person("John", "Waria", "FELLOW", "Y")
+        self.assertTrue([{'John WariaUID0': 'John Waria'}]
+                        == self.amity.fellow_info)
+
+    def test_staff_data_persisted(self):
+        self.amity.add_person("John", "Waria", "STAFF", "N")
+        self.assertTrue([{'John WariaUID0': 'John Waria'}]
+                        == self.amity.staff_info)
+
+    def test_adding_person_with_number_as_name(self):
+        self.assertTrue(self.amity.add_person(2, "Upendo", "STAFF", "N")
+                        == "Name cannot be a number!")
+
+    def test_addingPeople_with_unrecognized_roles(self):
+        self.assertTrue(self.amity.add_person("John", "Maasai", "Member", "Y")
+                        == "Role can only be STAFF or FELLOW")
+
+    def test_wrong_accomodation_option(self):
+        self.assertEqual(self.amity.add_person("Paul", "Upendo", "STAFF", "J"),
+                         "Accomodation options are only 'Y' or 'N'")
+
+    def test_if_staff_be_accomodated(self):
+        self.assertEqual(self.amity.add_person("Paul", "Upendo", "STAFF", "Y"),
+                         "Staff cannot have accomodation!")
+
+    """ Allocation and Reallocation"""
+
+    def test_reallocating_person_without_a_room(self):
+        self.assertEqual(self.amity.rellocate_person("UID002", "Hogwarts"),
+                         "This person was not allocated a room previously.")
+
+    def test_Reallocation_Success(self):
+        self.assertEqual(self.amity.rellocate_person("UID001", "Hogwarts"),
+                         "Person successfully reallocated")
+
+    def test_reallocating_to_non_existent_room(self):
+        self.assertEqual(self.amity.rellocate_person("UID001", "Narnia"),
+                         "Oops sorry, this particular room name does not exist!")
+
+    def test_reallocation_of_non_existent_person(self):
+        self.assertEqual(self.amity.rellocate_person("XCDEE", "Narniaa"),
+                         "Ooops, incorrect Person ID format or doesn't exist!")
+
+    def test_reallocation_to_same_room(self):
+        self.assertEqual(self.amity.rellocate_person("UID001", "Hogwarts"),
+                         "Ooops Upendo is already allocated to Hogwarts")
+
+    def test_reallocating_staff_to_livingSpace_fails(self):
+        self.amity.create_room("Egypt", "L")
+        self.amity.add_person("PAUL", "STAFF", "N")
+        self.assertTrue(self.amity.rellocate_person("STF001", "Egypt") ==
```

Comment:
use `self.assertEqual`. `self.assertTrue` and `self.assertFalse` is more appropriate for function that return boolean values